### ISSUE TITLE
Deployment template validation failed: 'Circular dependency detected on resource: '/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rgName/providers/Microsoft.ServiceFabric/clusters/clustername'

### ISSUE THREAD

**Issue by ORyabtsev** [Azure/service-fabric-issues#1671]
Hi,

I am trying to deploy via Azure DevOps Service Fabric Cluster with 2 node types using ARM Templates, but I get an error as below

2020-04-23T11:59:53.5606613Z There were errors in your deployment. Error code: InvalidTemplate.
2020-04-23T11:59:53.5610860Z ##[error]Deployment template validation failed: 'Circular dependency detected on resource: '/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rgName/providers/Microsoft.ServiceFabric/clusters/clustername'. 

Please see https://aka.ms/arm-template/#resources for usage details.'.
2020-04-23T11:59:53.5615780Z ##[warning]Validation errors were found in the Azure Resource Manager template. This can potentially cause template deployment to fail. Task failed while creating or updating the template deployment. Please follow https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-syntax

I even tried to remove all dependsOn in all resources in ARM template, I get the same error. What might be wrong with my template? The template is attached.

Any help will be appreciated!
[sf_cluster2.zip](https://github.com/Azure/service-fabric-issues/files/4522568/sf_cluster2.zip)

**Comment by ORyabtsev** [comment]
Any updates on my issue?

**Comment by ORyabtsev** [comment]
Can't anybody to assist with it? Trying to solve it, but can't. Anyone's help will be helpful. Thanks!

**Comment by ORyabtsev** [comment]
Hi again. As I see nobody took a look at my issue. Anyway, I resolved it. So, what I did. I started from scratch, deploying SF Cluster with one Node Type in the beginning. After successful deployment, I added the second Node Type with additional Load Balancer, Public IP Address, Network Security Group. After that, I've got a few other errors, but no more error regarding Circular dependency. Now, I have successfully deployed SF Cluster with two Node Types, Primary and Secondary. So, my advice to everybody who will face such an issue, sometimes it is worth starting from scratch if you tried all solutions which won't help. Good luck!
